reflex/state.py

```diff
@@ -3475,6 +3502,19 @@ async def _wait_lock(self, lock_key: bytes, lock_id: bytes) -> None:
                         break
                 state_is_locked = await self._try_get_lock(lock_key, lock_id)
 
+    @override
+    async def disconnect(self, token: str):
```

Comment:
This is probably wrong. Clients can reconnect. Need to check if there are any unexpired redis tokens

Reply:
i had a draft comment on my in progress review calling out this function, but i suppose your testing found it first. we need to keep the in memory states around. we do have an internal bug to add an expiry for memory state manager, but it just hasn't been high priority because we assume no one is using memory state manager in production

Reply:
Maybe one asyncio Lock is enough, we wont need to garbage collection then

Reply:
Note to myself: subscribe to redis expire events to clean up locks, may be easier with #4459